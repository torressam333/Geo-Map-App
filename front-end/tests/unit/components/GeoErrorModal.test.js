import { shallowMount } from '@vue/test-utils';
import GeoErrorModal from '@/components/GeoErrorModal';

describe('GeoErrorModal', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallowMount(GeoErrorModal, {
      propsData: {
        errorMessage: '',
      },
    });
  });

  it('emits a method to close the modal', async () => {
    wrapper.vm.$emit('closeGeoError');

    // Wait until $emits have been handled
    await wrapper.vm.$nextTick();

    expect(wrapper.emitted().closeGeoError).toBeTruthy();
  });

  it('renders the modal markup correctly', () => {
    expect(wrapper.element).toMatchSnapshot();
  });
});
