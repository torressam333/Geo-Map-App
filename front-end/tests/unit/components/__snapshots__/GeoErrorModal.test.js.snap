// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GeoErrorModal renders the modal markup correctly 1`] = `
<div
  class="h-screen absolute w-full z-10 flex justify-center items-start pt-[150px] bg-black/50"
>
  <div
    class="flex flex-col bg-white w-[80%] sm:w-[500px] px-6 py-8 rounded-md"
  >
    <h1
      class="text-xl mb-3"
    >
      Error: 
    </h1>
    <p
      class="text-sm mb-6"
    >
       This error may be caused because your location services may be disabled. Please enable locaiton services to make the maps work properly. 
    </p>
    <button
      class="self-end py-2 px-4 bg-red-600 text-white rounded-md text-sm"
    >
       Close Modal 
    </button>
  </div>
</div>
`;
